openmrs/openmrs-core
# TRUNK-4673 Add support for init parameters for module servlets

## Description of what I changed



This PR introduces the ability for modules to define servlet initialization parameters within their config.xml. The key changes and enhancements include:

- **Support for Initialization Parameters**: Modules can now define initialization parameters for servlets in their config.xml.
- **DTD Updates**: The Document Type Definition (DTD) has been updated to accommodate the changes and provide validation.
- **Testing Improvements**:
  - **Local DTDs for Testing**: Previously, our DTD tests relied on the server's DTDs. This approach posed challenges, especially when needing to update both the DTD and its corresponding tests in a single PR. This PR transitions the tests to use the DTDs stored in version control, ensuring consistency and ease of updates.
  - **New Tests**: Tests have been added to:
    - Verify the addition of initialization parameters to servlets.
    - Confirm the correctness of the DTD changes.

## Issue I worked on




see https://issues.openmrs.org/browse/TRUNK-4673

## Checklist: I completed these to help reviewers :)


- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command


  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`
